Python/ceval.c

```diff
@@ -6319,6 +6319,20 @@ format_exc_check_arg(PyThreadState *tstate, PyObject *exc,
         return;
 
     _PyErr_Format(tstate, exc, format_str, obj_str);
+
+    if (exc == PyExc_NameError) {
+        // Include the name in the NameError exceptions to offer suggestions later.
+        _Py_IDENTIFIER(name);
+        PyObject *type, *value, *traceback;
+        PyErr_Fetch(&type, &value, &traceback);
+        PyErr_NormalizeException(&type, &value, &traceback);
```

Comment:
Is it needed to normalize the exception? _PyErr_Format() should always create value, no?

Reply:
👍

Reply:
IIRC we need to normalize it, otherwise, it will not be set in the value field. Try to not do it and the test won't pass